models/actions/artifact.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+// This artifact server is inspired by https://github.com/nektos/act/blob/master/pkg/artifacts/server.go.
+// It updates url setting and uses ObjectStore to handle artifacts persistence.
+
+package actions
+
+import (
+	"context"
+	"fmt"
+
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/modules/timeutil"
+	"code.gitea.io/gitea/modules/util"
+)
+
+const (
+	// ArtifactUploadStatusPending is the status of an artifact upload that is pending
+	ArtifactUploadStatusPending = -1
+	// ArtifactUploadStatusConfirmed is the status of an artifact upload that is confirmed
+	ArtifactUploadStatusConfirmed = 1
+)
+
+func init() {
+	db.RegisterModel(new(ActionArtifact))
+}
+
+// ActionArtifact is a file that is stored in the artifact storage.
+type ActionArtifact struct {
+	ID               int64              `xorm:"pk autoincr"`
+	JobID            int64              `xorm:"index"`
+	RunnerID         int64              `xorm:"index"`
+	RepoID           int64              `xorm:"index"`
+	OwnerID          int64              `xorm:"index"`
+	CommitSHA        string             `xorm:"index"`
+	StoragePath      string             // The path to the artifact in the storage
+	FileSize         int64              // The size of the artifact in bytes
+	FileGzipSize     int64              // The size of the artifact in bytes after gzip compression
+	ContentEncnoding string             // The content encoding of the artifact
+	ArtifactPath     string             // The path to the artifact when runner uploads it
+	ArtifactName     string             // The name of the artifact when runner uploads it
+	UploadStatus     int64              `xorm:"index"` // The status of the artifact upload
+	Created          timeutil.TimeStamp `xorm:"created"`
+	Updated          timeutil.TimeStamp `xorm:"updated index"`
```

Comment:
Do we really need so many indexes? I mean there are only `ListArtifactByJobID` and `ListArtifactsByRepoID`.